preparedata/src/main/groovy/org/orbisgis/orbisprocess/geoclimate/preparedata/bdtopo/BDTopoGISLayers.groovy

```diff
@@ -112,7 +112,37 @@ IProcess importPreprocess(){
             def tmp_imperv_surface_activite = 'TMP_IMPERV_SURFACE_ACTIVITE_' + uuid
             def input_impervious = 'INPUT_IMPERVIOUS'
 
-            // If the impervious tables does not exists, we create corresponding empty tables
+
+            // -------------------------------------------------------------------------------
+            // Check if the input files are present
+
+            // If the IRIS_GE table does not exist or is empty, then the process is stopped
+            if(!datasource.hasTable(tableIrisName) || datasource.getTable(tableIrisName).isEmpty()){
+                logger.error 'The process has been stopped since the table IRIS_GE does not exist or is empty'
+                return null}
+
+            // If the following tables does not exists, we create corresponding empty tables
+            if(!datasource.hasTable(tableBuildIndifName)){
+                datasource.execute("CREATE TABLE $tableBuildIndifName (THE_GEOM geometry(geometry, 2154), ID varchar, HAUTEUR integer);")
```

Comment:
Custom SRID ???
What about if the user wants to process out of 2154 ?

Reply:
The idea was from you ;-)
Three options :
- we let as it is
- we remove the SRID
- the SRID is added as an input parameter

Reply:
From me to use one SRID ? my bad.
I think the SRID must be checked from the input table.

Reply:
Ok. And if the user has no .PRJ file or if the input table has no SRID ? We don't do anything ?

Reply:
Hum, the problem is more complex.
How to check the SRID on a table that does not exist (since this is the problem here -> if there is no table, then we want to create a corresponding empty one).

So, we have these options:
- check the SRID from other tables in the same folder (or schema)
    - but what do we do if 
        - there is different SRID ? Take the first one ? Set a default one ?
        - there is no SRID ? --> Force with a default one ?
- define the SRID as an input parameter and force all the files (tables) with it
    - if there is existing other SRID, do we reproject ?
    - if there is files (tables) with no PRJ (SRID), forcing with a default SRID may induce errors (e.g if the input files is in WGS84 (with no PRJ - so we don't have this information) and we force with 2154, it's not correct)
- explain to users that in their input folder (or schema), all the files (tables) have to have a same SRID --> we use the first SRID found

Reply:
One other option, after a short discussion with @j3r3m1, could be :

The user has to provide same PRJ files.

Then we make the following test :
```
IF all input files (or tables) have a PRJ file (SRID)
  THEN (IF the PRJ (SRID) are the same 
                  THEN we use the SRID and continue the process
                  ELSE error message and stop the process
               )
  ELSE error message and stop the process
```

Reply:
User must provide a valid PRJ that contains an EPSG code.

Reply:
So
Read the first table - check SRID, table.srid.
If SRID =0 then process without SRID
else check if the SRID is the same for all tables else throw exception

Reply:
Ok I will work on it

Reply:
Done.
Check the SRID of tables. 
- If all the SRID are the same, then use the SRID and process
- If differences, then stop the process.

In the doc, we will explain that .PRJ files are expected and have to be the same